core/src/main/scala/kafka/server/metadata/MetadataPartitions.scala

```diff
@@ -204,9 +248,10 @@ class MetadataPartitionsBuilder(val brokerId: Int,
   }
 
   def build(): MetadataPartitions = {
-    val result = MetadataPartitions(newNameMap, newIdMap)
+    val result = new MetadataPartitions(newNameMap, newIdMap, newReverseIdMap)
     newNameMap = Collections.unmodifiableMap(newNameMap)
```

Comment:
just curious, why we don't wrap values of `newNameMap` to immutable map? It seems `MetadataPartitionsBuilder#set(MetadataPartition)`is able to cause change after making metadata image.

Reply:
That's a fair point. Maybe it would be better to raise an exception in `set` after the image had been built.